solr/core/src/java/org/apache/solr/update/processor/AtomicUpdateDocumentMerger.java

```diff
@@ -83,6 +86,17 @@ public static boolean isAtomicUpdate(final AddUpdateCommand cmd) {
     
     return false;
   }
+
+  /**
+   *
+   * @param sdoc doc to be updated
+   * @param rootDoc the root document
+   * @return whether sdoc's id equals to rootDoc's id
+   */
+  public boolean isRootDoc(SolrInputDocument sdoc, SolrInputDocument rootDoc) {
```

Comment:
This method's signature is so weird and its implementation is short enough that I wonder if you could simply just inline it and we'd be happier?

Reply:
> * Can solrconfig-block-atomic-update.xml be deleted?

Yes :+1: 

>     * I didn't notice where you are using/testing the signatureField; I only noticed your edit to add the schema portion.

I started using solrconfig-tlog.xml, which uses the signature field URP.
It is now required that the schema have the required signatureFields.